### PR TITLE
Rename supplier to retailer + documentation

### PR DESCRIPTION
**Pull request by sdlumpp**
Documentation Chapter 1
Renamed supplier to retailer

### CODE REVIEW

File: lemlab/agents/prosumer.py [hunk 1]
```diff
@@ -997,7 +997,7 @@ def market_agent(self, db_obj, clear_positions=False):
                               + 15*60*self.config_dict["ma_horizon"])]
             [f"power_{self.config_dict['id_meter_grid']}"]) / 4
 
-        df_potential_bids.rename(columns={f"power_{self.config_dict['id_meter_grid']}": "net_bids"}, inplace=True)
+        df_potential_bids.rename(columns={f"power_{self.config_dict['R']}": "net_bids"}, inplace=True)
```
Comment (TUM-Doepfert) [inline lemlab/agents/prosumer.py:1000]:
```suggestion
        df_potential_bids.rename(columns={f"power_{self.config_dict['id_meter_grid']}": "net_bids"}, inplace=True)
```
;)

**Review by TUM-Doepfert — CHANGES_REQUESTED** [review]
Please see comments.

File: docs/01_Introduction_to_LEMs.rst [hunk 2]
```diff
@@ -3,56 +3,119 @@ Introduction to LEMs
...
+fluctuating renewables) as well as the anticipated electrification of the mobility and heat sectors in the form of
+electric vehicles and heat pumps.
+
+The European Union has included new rules in its "Directive on common rules for the internal electricity market" ((EU)
+2019/944) "that enable active consumer participation, individually or through citizen energy communities, in all
+markets, either by generating, consuming, sharing or selling  electricity, or by providing flexibility services through
+demand-response and storage.  The directive aims to improve the uptake of energy communities and make it easier for
+citizens to integrate efficiently in the electricity system, as active participants."
+
+Specifically, the aim of these energy communities it to enable decentralized producers and consumers (prosumers) to
```
[the remark this answers is not in the document]
Comment (sdlumpp) [reply docs/01_Introduction_to_LEMs.rst:22]: Accepted.

File: docs/01_Introduction_to_LEMs.rst [hunk 3]
```diff
@@ -3,56 +3,119 @@ Introduction to LEMs
...
+.. image:: images/lem_structure.png
+    :width: 400px
+    :align: center
+    :height: 263px
+    :alt: Architecture of a generalized LEM
+
+At the heart of the *lemlab* is the **prosumer**. In modern power systems, the end user is no longer just a consumer.
+The prosumer is the generalized market participant, combining production, consumption, and flexibility with a
+preference for local and sustainable energy. Of course a prosumer could be a pure consumer, a simple producer, they
+might be entirely inflexible or they might prefer his electrical energy sources to produce as much carbon dioxide as
```
[the remark this answers is not in the document]
Comment (sdlumpp) [reply docs/01_Introduction_to_LEMs.rst:51]: 1. accepted.
2. intentionally tongue in cheek ;)

File: docs/01_Introduction_to_LEMs.rst [hunk 4]
```diff
@@ -3,56 +3,119 @@ Introduction to LEMs
...
+possible. In any case, we will still refer to them as a prosumer.
+
+Prosumers place bids on a **LEM platform**. This platform combines all market functionality, beginning with user
```
[the remark this answers is not in the document]
Comment (sdlumpp) [reply docs/01_Introduction_to_LEMs.rst:54]: Rejected, grammatically correct the way it is.

File: docs/01_Introduction_to_LEMs.rst [hunk 5]
```diff
@@ -3,56 +3,119 @@ Introduction to LEMs
...
+management, bid/offer collection and clearing, and ending with meter reading collection, market settlement and levy
+collection. **Aggregators** trade on a LEM on behalf of a group of prosumers. **Retailers** take care of coupling the
```
[the remark this answers is not in the document]
Comment (sdlumpp) [reply docs/01_Introduction_to_LEMs.rst:56]: Rejected, see above.

File: docs/01_Introduction_to_LEMs.rst [hunk 6]
```diff
@@ -3,56 +3,119 @@ Introduction to LEMs
...
+LEM to the wholesale market.
+
+That's pretty much it. Of course we can also model grid operators, transmission grids and wholesale markets as well
+as multiple LEMs. These are possible in *lemlab* and some are covered in this documentation. All are simply a matter of
+rearranging the building blocks *lemlab* supplies to construct the desired system.
+
+When does what happen in a LEM?
```
[the remark this answers is not in the document]
Comment (sdlumpp) [reply docs/01_Introduction_to_LEMs.rst:63]: Rejected, see above.

File: docs/01_Introduction_to_LEMs.rst [hunk 7]
```diff
@@ -3,56 +3,119 @@ Introduction to LEMs
...
+    :height: 150px
+    :alt: Basic timeline for a generalized LEM
+
+We divide our market timeline into four fundamental time periods. The first is **registration**, which takes place
+before trading commences. We then enter the **rolling horizon market sequence**, which repeats for each delivery period.
+Each day is divided into an arbitrary number of periods of energy exchange, referred to as **timestep of delivery**
+(ts_delivery) labelled as the period between td and td + T. We usually use 15 minute windows (T=900s), as the German
+electricity market is divided into 15 minute periods. During this time, the physical flow of energy takes place. All
+other activities either take place between market opening, td - tO, and the beginning of the ts_delivery in question,
+td, and are known as **pre-exchange-activities**, or they take place after delivery, which case they are known as
```
[the remark this answers is not in the document]
Comment (sdlumpp) [reply docs/01_Introduction_to_LEMs.rst:81]: Accepted.

File: docs/01_Introduction_to_LEMs.rst [hunk 8]
```diff
@@ -3,56 +3,119 @@ Introduction to LEMs
...
+**settlement** activities. Examples of pre-exchange activities are forecasting of demand and energy trading, while
+settlement activites include meter value logging, market settlement and balancing energy settlement.
+
+.. image:: images/lem_timeline_detailed.png
+    :width: 450px
+    :align: center
+    :height: 204px
+    :alt: Detailed timeline for a generalized LEM
+
+The above figure shows a slightly more detailed timeline. We see pre-clearing activities taking place during each of *n*
```
[the remark this answers is not in the document]
Comment (sdlumpp) [reply docs/01_Introduction_to_LEMs.rst:91]: Perhaps. Will be discussed during the paper writing process.

File: lemlab/agents/prosumer.py (hunk 1 above)
Comment (sdlumpp) [reply lemlab/agents/prosumer.py]: Resolved.

**Review by TUM-Doepfert — APPROVED** [review]
Test case run succesful

**Pull request merged**